deposit/src/main/java/edu/unc/lib/deposit/staging/StagingPolicyManager.java

```diff
@@ -0,0 +1,146 @@
+ */
+package edu.unc.lib.deposit.staging;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.codehaus.jackson.map.ObjectMapper;
+
+import edu.unc.lib.deposit.staging.StagingPolicy.CleanupPolicy;
+import edu.unc.lib.dl.util.URIUtil;
+
+/**
+ * Manager for staging policies, which assists in determining if file paths are
+ * located within accepted staging locations according to the provided policies,
+ * and provides access to those policies. Loaded from a policy configuration
+ * json file
+ * 
+ * @author bbpennel
+ *
+ */
+public class StagingPolicyManager {
+
+	private String basePath;
+
+	private String configPath;
+
+	private List<StagingPolicy> policies;
+
+	public StagingPolicyManager() {
+	}
+
+	public void init() {
+		loadConfig();
+	}
+
+	private void loadConfig() throws StagingException {
+		File configFile = new File(configPath);
+
+		ObjectMapper mapper = new ObjectMapper();
+
+		try {
+			// Load the policy configuration from the provided json
+			policies = mapper.readValue(configFile, mapper.getTypeFactory()
+					.constructCollectionType(List.class, StagingPolicy.class));
+		} catch (IOException e) {
+			throw new StagingException("Failed to load staging configuration from " + configPath, e);
+		}
+
+		// Verify and resolve any staging location paths
+		for (StagingPolicy policy : policies) {
+			String path = policy.getPath();
+
+			// If path was no absolute, resolve it against the base path
+			if (!Paths.get(path).isAbsolute()) {
+				policy.setPath(URIUtil.join(basePath, path));
+			}
+
+			// Abort loading if one of the paths is invalid
+			if (!Paths.get(policy.getPath()).toFile().exists()) {
+				throw new StagingException("Unable to resolve staging location " + policy.getPath());
+			}
+		}
+	}
+
+	/**
+	 * Get the staging policy which applies to the given file uri, or throw a
+	 * staging exception if no policy is found
+	 * 
+	 * @param fileUri
+	 * @return
+	 * @throws StagingException
+	 */
+	public StagingPolicy getStagingPolicy(URI fileUri) throws StagingException {
+		String path = fileUri.getPath();
+
+		for (StagingPolicy policy : policies) {
+			String policyPath = policy.getPath();
+			if (path.startsWith(policyPath)) {
+				return policy;
+			}
+		}
+
+		throw new StagingException("No staging policy available for " + fileUri);
+	}
+
+	/**
+	 * Get the cleanup policy which applies to the given file uri, or throw a
+	 * staging exception if no policy is found
+	 * 
+	 * @param fileUri
+	 * @return
+	 * @throws StagingException
+	 */
+	public CleanupPolicy getCleanupPolicy(URI fileUri) throws StagingException {
+		return getStagingPolicy(fileUri).getCleanupPolicy();
+	}
+
+	/**
+	 * Returns true if the given file uri is contained by at least one of the
+	 * staging locations registered with this manager
+	 * 
+	 * @param fileUri
+	 * @return
+	 */
+	public boolean inValidStagingLocation(URI fileUri) {
```

Comment:
yeah that is pretty unfortunate isn't it.  I think i'll change it to isValidStagingLocation